esmf_regrid/experimental/unstructured_scheme.py

```diff
@@ -70,7 +72,7 @@ def _cube_to_GridInfo(cube):
 #     return result
 
 
-def _create_cube(data, src_cube, mesh_dim, mesh, grid_x, grid_y):
+def _create_cube(data, src_cube, mesh_dim, grid_x, grid_y):
```

Comment:
Originally this followed a pattern seen for example [here](https://github.com/SciTools/iris/blob/f76288cb82eefc5bc3c74461bfd8f397618eda14/lib/iris/analysis/_regrid.py#L819). In this pattern, (`x_dim`, `y_dim`) is equivalent to `mesh_dim` and (`src_x_coord`, `src_y_coord`) is equivalent to `mesh`. If we followed the same pattern then the purpose of `mesh` would be to determine which coordinates to copy over from `src_cube` to the new cube, this would be done by performing a comparison which used `mesh`. We anticipate that this would be done in multidimensional regridding. However, since a cube can only have one mesh, all that will need to be checked is whether the coord is a MeshCoord (or has a mesh attribute). It is therefore unlikely we will need `mesh` during cube creation and so I am removing it from the arguments.